### ISSUE TITLE
Do not elide lines with `bun run` with workspaces

### ISSUE THREAD

**Issue by spiffytech** [oven-sh/bun#12431]
### What is the problem this feature would solve?

My DB migration library prints out what's wrong with my SQL, but running `bun run migrate` from my project top-level omits the most important lines from the error: the SQL error message.

```
$ bun run migrate
$ bun run --filter server --env-file .env migrate
server migrate $ bunx pg-migrations apply --directory migrations && bunx @databases/pg-schema-cli --directory __generated__
│ [8 lines elided]
│     at handleError (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg/lib/Driver.js:394:29)
│     at executeQueryInternal (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg/lib/Driver.js:369:9)
│     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
│     at async PgDriver._executeQuery (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg/lib/Driver.js:199:29)
│     at async PgDriver.executeAndReturnLast (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg/lib/Driver.js:226:16)
│     at async Transaction.query (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/shared/lib/BaseTransaction.js:59:24)
│     at async /home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg-migrations/lib/PostgresDatabaseEngine.js:19:25
│     at async Object.applyMigration (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg-migrations/lib/PostgresDatabaseEngine.js:137:21)
│     at async /home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/migrations-base/lib/methods/applyOperations.js:43:29
│     at async /home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg-migrations/lib/PostgresDatabaseEngine.js:54:20
└─ Exited with code 1
error: script "migrate" exited with code 1
```

If I `cd` into `packages/server` and run the same command manually I get the full error:

```
$ bun run --env-file ../../.env migrate
$ bunx pg-migrations apply --directory migrations && bunx @databases/pg-schema-cli --directory __generated__
Applying 0003-lifts-order.sql
error: column lifts.id does not exist

  16 |   ) as subquery
  17 | where
> 18 |   lifts.id = subquery.id
     |   ^

    at handleError (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg/lib/Driver.js:394:29)
    at executeQueryInternal (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg/lib/Driver.js:369:9)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async PgDriver._executeQuery (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg/lib/Driver.js:199:29)
    at async PgDriver.executeAndReturnLast (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg/lib/Driver.js:226:16)
    at async Transaction.query (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/shared/lib/BaseTransaction.js:59:24)
    at async /home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg-migrations/lib/PostgresDatabaseEngine.js:19:25
    at async Object.applyMigration (/home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg-migrations/lib/PostgresDatabaseEngine.js:137:21)
    at async /home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/migrations-base/lib/methods/applyOperations.js:43:29
    at async /home/spiffytech/Documents/bootspoon/barbellbasics/node_modules/@databases/pg-migrations/lib/PostgresDatabaseEngine.js:54:20
error: script "migrate" exited with code 1
```

### What is the feature you are proposing to solve the problem?

A way for `bun run` to print the entire, unabridged output from workspace scripts. Maybe by default, maybe with a flag.

### What alternatives have you considered?

Just not using top-level `package.json` scripts if they're not gonna show me complete output, but then I have to specify the `.env` file for every command

**Comment by ChristopherHaws** [comment]
I wouldn't mind it not being the default, but I would love to be able to stream the output. PNPM has a flag for this:

```
--stream   : Stream output from child processes immediately, prefixed with the originating package directory. This allows output from different packages to be interleaved.
```

**Comment by chneau** [comment]
One I have a good user experience with is by using `concurrently` npm package with the `--names` to add prefix names, the `--prefix-colors=auto` for auto coloration.

Here is an example, left is `bun run --filter '*' start`, right if `concurrently --names=client,server --prefix-colors=auto 'bun run start:client' 'bun run start:server'`

![image](https://github.com/user-attachments/assets/5a9718b3-a199-4f9d-9b38-2f64627c5478)

**Comment by nikitavoloboev** [comment]
I hope bun team looks into this, it's quite an annoying blocker of using this feature with bun.

**Comment by erik-reges** [comment]
please provide us a with a way of running bun applications that actually shows terminal output as a whole and not just the 5 latest lines

**Comment by gokaygurcan** [comment]
I think I have two issues in one: 
- elided rows preventing to see the full output
- bun blocking the terminal inputs, and preventing me to make any selection

```bash
$ bun --filter="./packages/database" db:push --force                                                                                                                                                  
@project/database db:push $ drizzle-kit push --config=drizzle.config.ts
│ [15 lines elided]                                                                  <--- first issue here
[⣽] Pulling schema from database...
[⣷] Pulling schema from database...
[⣯] Pulling schema from database...
[⣟] Pulling schema from database...
[⡿] Pulling schema from database...
[✓] Pulling schema from database...
│ 
│ Is customer_id column in subscriptions table created or renamed from another column?
│ ❯ + customer_id              create column                                         <--- second issue here
│   ~ customerId › customer_id rename column
└─ Done in 12.82 s
```

Any luck with the first part? How can we see the full output instead of last few lines? And, anyone having a similar problem with blocked inputs?

**Comment by dsculptor** [comment]
This works, but your program may lose terminal abilities as well.

```sh
NO_COLOR=1 bun ...
```

**Comment by suhascv7-nombolo** [comment]
This is annoying during development, hope bun fixes this soon

**Comment by IvanLi-CN** [comment]
This was a nice feature until my program reported an error and the valuable error message was completely invisible.

This issue should be taken seriously.

**Comment by ScottMorse** [comment]
This should really be addressed. I'd like to use bun workspaces but place scripts in my root package.json that filter for specific packages, e.g. `"my-app": "bun --filter '@my/app'"` so that one can run `bun run my-app test` etc., making the root project's scripts like a simple CLI. However, the worst commands  to run this way are `test` and `lint` etc., because there often is no way to see the most important output of these. Using `NO_COLOR=1` helps, but often test/lint commands benefit greatly from the use of terminal colors, so this is unideal.

**Comment by ScottMorse** [comment]
For what it's worth, I've created a simple CLI called [bun-workspaces](https://github.com/ScottMorse/bun-workspaces#readme) to help with running commands from a workspace.  This lets you run a script from your project root via the workspace's package.json name, but the lines won't get elided since under the hood, the commands are ran from the respected workspace's root instead of via `--filter`.

Some examples:
```bash
alias bw="bunx bun workspaces"

# Run my-script for all
# workspaces that include it
# in their `scripts` field
bw run my-script

# Run my-script for a specific workspace
bw run my-script my-workspace

# Run my-script for multiple specific workspaces
bw run my-script workspace-a workspace-b

# Run my-script in parallel for all workspaces
bw run my-script --parallel

# Append args to each script call
bw run my-script --args "--my --args"
```

You can also get metadata about the workspaces, such as  via `bunx bun-workspaces list-workspaces`. This is like having a lightweight Nx-like CLI on top of vanilla workspaces.

This is hot out of the oven, so extremely new and experimental, but I did give it some test coverage and will be using it, since I wrote it for myself, unless the Bun CLI improves the features in this regard.

**Comment by martinamps** [comment]
Put up a PR to make this overridable, hopefully makes it through! 🙏

**Comment by Jarred-Sumner** [comment]
Fixed by @martinamps in #15837 

This will be part of the Bun v1.1.43 release

**Comment by nerdo** [comment]
This is even more critical with AI workflows these days. Having llms able to clearly see what's going on with servers is helpful. Hiding it by default is not.
